qa/paqa_devs.c

```diff
@@ -118,7 +118,7 @@ static int gNumFailed = 0;
 static float NextSineSample( PaQaData *data )
 {
     float phase = data->phase + PHASE_INCREMENT;
-    if( phase > M_PI ) phase -= 2.0 * M_PI;
+    if( phase > M_PI ) phase -= 2.0 * (float) M_PI;
```

Comment:
This doesn't look right `2.0` is a double, so you're multiplying a double by `(float) M_PI` and you will get a double.

```suggestion
    if( phase > M_PI ) phase -= (float)(2.0 * M_PI);
```

The double precision multiply should happen at compile time.

Reply:
Good catch. I'm a bit confused as to why my change got rid of the warning then. Maybe C4305 is only produced with some kinds of expressions but not others, I don't know.